install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml

```diff
@@ -0,0 +1,18 @@
+---
+- name: Install dependencies for building translations
+  apt:
+    pkg: "{{ item }}"
+    state: present
+  with_items: "{{ appserver_dependencies }}"
+  tags: apt
+
+- name: Install pip dependencies for SecureDrop.
+  pip:
+    requirements: "{{ securedrop_code_filtered }}/requirements/securedrop-requirements.txt"
+  tags:
+    - pip
+
+- name: Compile PO to MO.
+  shell: "cp config.py.example config.py ; trap 'rm config.py' EXIT ; PYTHONDONTWRITEBYTECODE=true ./manage.py --verbose translate --compile"
+  args:
+    chdir: "{{ securedrop_code_filtered }}"
```

Comment:
Crafty! Hard to read, though. Seems like the desired control flow is:

1. Ensure that `config.py` exists.
2. Run the `manage.py translate --compile` command.
3. Remove the `config.py` file.

We could definitely break this up into separate tasks to help readability, but perhaps the functionality would be clear to future maintainers. Still, a reformatting of the current logic like so:

```
diff --git a/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml b/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml
index 0c9a574d..a4653733 100644
--- a/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml
+++ b/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml
@@ -13,6 +13,11 @@
     - pip
 
 - name: Compile PO to MO.
-  shell: "cp config.py.example config.py ; trap 'rm config.py' EXIT ; PYTHONDONTWRITEBYTECODE=true ./manage.py --verbose translate --compile"
+  shell: >-
+    cp config.py.example config.py ;
+    trap 'rm config.py' EXIT ;
+    ./manage.py --verbose translate --compile
   args:
     chdir: "{{ securedrop_code_filtered }}"
+  env:
+    PYTHONDONTWRITEBYTECODE: true
```

Strikes me as an improvement. (Note I haven't actually tested the patch above.)

Reply:
Much better to read indeed, thanks for the diff :-)